pyprep/utils.py

```diff
@@ -362,6 +362,44 @@ def split_list(mylist, chunk_size):
     ]
 
 
+def print_progress(current, end, start=None, stepsize=1, every=0.1):
```

Comment:
why did you opt to code up you own progress tracking function instead of using [`tqdm`](https://github.com/tqdm/tqdm)? To avoid an additional dependency?

Reply:
It'd be fun to see how fast/slow your function is compared to what tqdm says:

> Overhead is low -- about 60ns per iteration (80ns with tqdm.gui), and is unit tested against performance regression. By comparison, the well-established ProgressBar has an 800ns/iter overhead.

Reply:
Huh. My guess is that their code is probably better optimized but that my thing is still faster than ProgressBar (which I also didn't know existed until just now): I made no attempt to optimize my own approach, but I don't think there are any particularly expensive or wasteful function calls I'm doing per loop and I'm not trying to write as much to the terminal on every update.